### PR TITLE
Use TOML for configuration

### PR DESCRIPTION
**Pull request by networkfusion**
## Description

To ensure a standard.

Raised as a draft as the code can probably be improved... but I wanted to share thoughts.

## Motivation and Context

Easier to read than YAML, and Json
Not as verbose as XML

## How Has This Been Tested?

On a local cart.

## Types of changes

- [ ] Improvement (non-breaking change that adds a new feature)
- [ ] Bug fix (fixes an issue)
- [x] Breaking change (breaking change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Signed-off-by: GITHUB_USER <GITHUB_USER_EMAIL>

### CODE REVIEW

File: src/menu/menu.c [hunk 1]
```diff
@@ -1,57 +1,103 @@
...
 #include "flashcart/flashcart.h"
+#include "libs/toml/toml.h"
 
 #include "menu.h"
 
 
-static char *config_path = "n64/config.txt";
-static char game_path[256];
-static char save_path[256];
+#define SC64_CONFIG_FILEPATH "sd://config.sc64.toml.txt"
```
Comment (bigbass1997) [inline src/menu/menu.c:14]: Why `.toml.txt`? The standard expects the file extension to be `.toml`, and I think people used to this format would also expect that.

Comment (networkfusion) [reply src/menu/menu.c:14]: for the moment (at least whilst in development) it is easier on windows to edit the file.
I would expect the extension and name/location to be changed before going gold.

**Comment by bigbass1997** [comment]
I think using TOML is a wonderful choice for a config file. Although I may be biased from my use of it in Rust environments, it is a good balance between other formats.

File: src/menu/menu.c [hunk 2]
```diff
@@ -1,57 +1,103 @@
...
+    char error_buffer[266];
+    
+    printf("Loading config file %s\n", SC64_CONFIG_FILEPATH);
+    wait_ms(1000);
+
+    fp = fopen(SC64_CONFIG_FILEPATH, "r");
+	if (!fp) {
+		printf("Error loading config file %s\n", SC64_CONFIG_FILEPATH);
+        wait_ms(10000);
+        assertf(!fp, "Couldn't open toml config file: %s", SC64_CONFIG_FILEPATH);
```
Comment (Polprzewodnikowy) [inline src/menu/menu.c:33]: Config loading function definitely shouldn't throw asserts and instead return error code. Menu then could present warning screen for the user if config is broken and default settings were used.

**Comment by networkfusion** [comment]
Current improvements made on this branch: https://github.com/networkfusion/N64FlashcartMenu/tree/develop-menu

**Pull request merged**